operator/cmd/mesh/operator-remove.go

```diff
@@ -78,13 +78,17 @@ func operatorRemove(cmd *cobra.Command, args *RootArgs, orArgs *operatorRemoveAr
 		l.LogAndFatal(err)
 	}
 
+	switch {
 	// If the user is performing purge but also specified a revision, an error message
 	// should be displayed and do nothing
-	if orArgs.purge && orArgs.revision != "" {
+	case orArgs.purge && orArgs.revision != "":
 		orArgs.revision = ""
 		l.LogAndFatal("At most one of the --revision or --purge flags could be set\n")
-	} else if orArgs.revision == "default" {
+	case orArgs.purge:
 		orArgs.revision = ""
+	case orArgs.revision == "":
```

Comment:
This is not correct, In other commands, we only set `default` to empty, why here set empty to `default`? This may cause some errors since the component may become something like `istiod-default` instead of `istiod`

Reply:
This is where it needs to work with the place commented above:

```
func isControllerInstalled(cs kubernetes.Interface, operatorNamespace string, revision string) (bool, error) {
	orn := operatorResourceName
	if revision != "" && revision != "default" {
```